Tasks/AzureCLIV1/azureclitask.ts

```diff
@@ -132,13 +132,24 @@ export class azureclitask {
         // set az cli config dir
         this.setConfigDirectory();
 
+        // set config.json 
+        this.loadServicePrincipalsIfRequired(servicePrincipalId, cliPassword, subscriptionID);
+
         //login using svn
         this.throwIfError(tl.execSync("az", "login --service-principal -u \"" + servicePrincipalId + "\" -p \"" + cliPassword + "\" --tenant \"" + tenantId + "\""), tl.loc("LoginFailed"));
         this.isLoggedIn = true;
         //set the subscription imported to the current subscription
         this.throwIfError(tl.execSync("az", "account set --subscription \"" + subscriptionID + "\""), tl.loc("ErrorInSettingUpSubscription"));
     }
 
+    private static loadServicePrincipalsIfRequired(servicePrincipalId, client_key, subscriptionID): void {
+        var loadFileName = tl.getVariable("LOAD_SERVICE_PRINCIPAL");
```

Comment:
We should not expect the user to specify the file name as the file name is kind of fixed in the az cli code and expect user to specify true/false only. 

Secondly LOAD word does not look right. We should change it to CREATE_SERVICE_PRINCIPAL_FILE

Reply:
What is the scenario for this requirement?

Reply:
This is required for supporting a scenario where the user wants to use the SPN details of the `serviceEndpoint` while creating a new AKS cluster, instead of creating a new service principal every time.

Reply:
Got it. It also addresses #8726 . Should the secrets rather be set as secret variables instead of this file for easier consumption? Also I believe this file will get deleted after the task exists (due to logout) - is that correct?

Reply:
This file is a convention followed by `az cli` which we figured out by following the source code. 
I noticed `az logout` doesn't explicitly delete that file, so I'm adding logic for unlinking upon exit.

Reply:
If this is the already saved by az cli - then why do we need to store it again?

Reply:
I meant, the file is a convention for the `az aks` module i.e. it searches for this file in the config path, otherwise tries to create a new service-principal for AKS cluster creation. 
This file is not available by default.

Reply:
Got it. Given that we have requests for use cases other than aks (as pointed above), and also this file syntax is an internal implementation of aks (subject to change) - should we consider the environment variables so that this feature is more usable for all scenarios.

Reply:
Azure endpoint can take client key or a certificate. Az CLI today does no support certificate yet (Azure PS does). @Ajay-MS would likely enhance Az CLI as well. Want to know if the solution proposed will scale for both client key and certificate?